Add distro compatibility policy (#3751)
moveit/moveit2: .github/DISTRO_COMPAT_POLICY.md

**Entry format**: date-prefixed bullet under the appropriate distro section. One sentence stating the change, one line of rationale (link the upstream PR that motivated it), and a diff block for non-obvious migrations.

```markdown
## ROS Rolling
- [06/2026] `position_controllers` was removed from `ros2_controllers` 6.7.0 ([ros-controls/ros2_controllers#2016](https://github.com/ros-controls/ros2_controllers/pull/2016)). Consumers using `position_controllers/JointGroupPositionController` should migrate to `forward_command_controller/ForwardCommandController` with `interface_name: position`. Consumers using `position_controllers/GripperActionController` should migrate to `parallel_gripper_action_controller/GripperActionController`.
```

**PR review checklist**: reviewers reject any PR that introduces a distro-conditional change without a MIGRATION.md entry.

**Cleanup**: distro release-note entries stay (they're historical). *Workaround* entries (e.g. "Remove when Resolute syncs osqp_vendor") get retired together with the workaround code.

---

## Topic 2: Handling dependencies without a bloom package

> When a moveit2 dependency is **released** for a distro (entry exists in `rolling/distribution.yaml`) but **not yet bloomed** to the apt binary repo, CI fails at `rosdep install`. Two complementary approaches handle this. **Prefer §2.1.**

Reviewer: When a package exists in rosdistro, it was bloomed. However, it might exist in ros-testing only, because it requires a sync. I suggest dropping the mention of bloom (here and below).
```suggestion
When a moveit2 dependency is not yet released for a distro, CI will fail at `rosdep install`. Two complementary approaches handle this. **Prefer §2.1.**
```

# Remove each entry when the corresponding ros-rolling-* deb appears at
# https://repo.ros2.org/status_page/ros_rolling_default.html
repositories:
  geometric_shapes:    # rolling release 2.3.3-2, awaiting Resolute sync
    type: git
    url: https://github.com/ros-planning/geometric_shapes.git
    version: 2.3.3-2
  osqp_vendor:        # rolling release 0.2.0-4
    type: git
    url: https://github.com/tier4/osqp_vendor.git
    version: 0.2.0-4
  stomp:              # rolling release 0.1.2-4
    type: git
    url: https://github.com/ros-industrial/stomp.git
    version: 0.1.2-4
```

### §2.2 — CMake optional + feature flag (fallback)

Reach for §2.2 only when §2.1 isn't viable — e.g. the upstream is private/license-restricted, build-broken, or the CI cost of source building is prohibitive.

When a dep is genuinely missing and the consuming code can degrade gracefully, use `QUIET` + a presence check. Always pair the optional `find_package` with a `target_compile_definitions` feature flag so source code can `#ifdef`-guard usages.

```cmake
# osqp_vendor not yet bloomed for Resolute; degrade gracefully when absent.
find_package(osqp QUIET)

> if(osqp_FOUND)
>   add_subdirectory(online_signal_smoothing)
>   target_compile_definitions(my_target PRIVATE MOVEIT_HAVE_OSQP)
> endif()

Reviewer: Add an `else` branch to inform users about the omitted feature (as required below).
```suggestion
if(osqp_FOUND)
  add_subdirectory(online_signal_smoothing)
  target_compile_definitions(my_target PRIVATE MOVEIT_HAVE_OSQP)
else()
  message(STATUS "Dropping online_signal_smoothing as osqp wasn't found")
endif()
```
